src/main/java/io/nats/client/JetStream.java

```diff
@@ -13,15 +13,91 @@
 
 package io.nats.client;
 
+import io.nats.client.api.ConsumerConfiguration;
 import io.nats.client.api.PublishAck;
+import io.nats.client.api.StorageType;
+import io.nats.client.api.StreamConfiguration;
 import io.nats.client.impl.Headers;
 
 import java.io.IOException;
 import java.time.Duration;
 import java.util.concurrent.CompletableFuture;
 
 /**
- * JetStream context for creation and access to streams and consumers in NATS.
+ * JetStream context for access to streams and consumers in NATS.
+ * 
+ * <h3>Basic usage</h3>
+ * 
+ * <p>{@link #publish(String, byte[]) JetStream.Publish} will send a message on the specified subject as a Req/Rep waiting for acknowledgement. 
+ * An error will be received if no stream is litetning on said subject.
+ * 
+ * <p>{@link Connection#publish(String, byte[]) Connection.Publish} can be used as well to send to a stream on a subject, but no confirmation or error will come back. 
+ * Use non acknowledged publish with caution as it is easy to overwhelm Jetstream this way.
+ * 
+ * <p>{@link #publishAsync(String, byte[]) PublishAsync} will not wait for acknowledgement but return a {@link CompletableFuture CompletableFuture}, 
+ * which can be checked for acknowledgement at a later point.
+ * 
+ * <p> Use {@link #getStreamContext(String ) getStreamContext(String)} for <b>consuming/subscribing</b> messages from Jetstream. 
```

Comment:
Maybe add something before this to mention "Simplified Consuming" and explain what it is.